### PR TITLE
fix: ensure that proxy logs are updated after the xhr has actually completed

### LINKED ISSUE #21361: Response body sometimes not displayed in console props for xhr request
### Current behavior

The response body is not always displayed in the console props for xhr requests. There appears to be a race condition between the xhr request finishing and the proxy sending the `response:received` event.

Proxy sends the `response:received` event:
https://github.com/cypress-io/cypress/blob/aa6f3b8039495ac37ced4ac581129a020ba66fa4/packages/proxy/lib/http/request-middleware.ts#L51-L57
Call made to update the console props after receiving the event:
https://github.com/cypress-io/cypress/blob/aa6f3b8039495ac37ced4ac581129a020ba66fa4/packages/driver/src/cypress/proxy-logging.ts#L336
Console props are updated with the xhr values. However, if the xhr isn't fully complete (i.e. readystate === DONE), then these values are not valid.
https://github.com/cypress-io/cypress/blob/aa6f3b8039495ac37ced4ac581129a020ba66fa4/packages/driver/src/cypress/proxy-logging.ts#L226-L231

This is causing flake in the following test:
Screen Shot 2022-05-05 at 4 50 41 PM

### Correct console props:
Screen Shot 2022-05-05 at 4 22 27 PM

### Incorrect console props:
Screen Shot 2022-05-05 at 4 21 02 PM

### Desired behavior

Console props always contains the xhr response body.

### Test code to reproduce

Run the following test multiple times until it fails:

https://github.com/cypress-io/cypress/blob/aa6f3b8039495ac37ced4ac581129a020ba66fa4/packages/driver/cypress/integration/cypress/proxy-logging_spec.ts#L153

### Cypress Version

develop

### Other

_No response_

### PR DESCRIPTION
**Pull request by ryanthemanuel**
- Closes #21361 

### User facing changelog

XHR calls will now properly display the response body in the command log

### Additional details

When the proxy gets the close event for the response, we send the `response:received` event to the client. There is a race condition where this might reach the client before the XHR readystate has been updated on the client. If this happens, the xhr that gets attached to the log will not have a response body yet. To fix this, we will listen to the `onload` event of the xhr call if the xhr call hasn't completed and delay updating the proxy logging response until then.

### How has the user experience changed?

n/a

### PR Tasks

- [X] Have tests been added/updated?
- [X] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/driver/src/cypress/proxy-logging.ts [hunk 1]
```diff
@@ -343,6 +337,29 @@ export default class ProxyLogging {
...
   }
 
+  private updateRequestWithResponse (responseReceived: BrowserResponseReceived): void {
+    const proxyRequest = _.find(this.proxyRequests, ({ preRequest }) => preRequest.requestId === responseReceived.requestId)
+
+    if (!proxyRequest) {
+      return debug('unmatched responseReceived event %o', responseReceived)
+    }
+
+    if (proxyRequest.xhr && proxyRequest.xhr.xhr.readyState !== 4) {
```
Comment (mschile) [inline packages/driver/src/cypress/proxy-logging.ts:347]:
```suggestion
    if (proxyRequest.xhr && proxyRequest.xhr.xhr.readyState !== XMLHttpRequest.DONE) {
```

File: packages/driver/src/cypress/proxy-logging.ts [hunk 2]
```diff
@@ -343,6 +337,29 @@ export default class ProxyLogging {
...
+      proxyRequest.xhr.xhr.onload = function (...args) {
+        if (oldOnload) {
+          oldOnload.call(this, ...args)
+        }
+
+        proxyLogging.updateProxyRequestWithResponse(proxyRequest, responseReceived)
+      }
+    }
+
+    this.updateProxyRequestWithResponse(proxyRequest, responseReceived)
```
Comment (mschile) [inline packages/driver/src/cypress/proxy-logging.ts:360]: Should this be in an `else` or do we need `updateProxyRequestWithResponse` called twice?

**Review by mschile — CHANGES_REQUESTED** [review]

File: packages/driver/src/cypress/proxy-logging.ts (hunk 2 above)
Comment (ryanthemanuel) [reply packages/driver/src/cypress/proxy-logging.ts]: Doh. That was an artifact of refactoring. Fixed here: [`70b07df` (#21373)](https://github.com/cypress-io/cypress/pull/21373/commits/70b07dfe4acb71861a97c01660c3017e82aae5b1)

File: packages/driver/src/cypress/proxy-logging.ts (hunk 1 above)
Comment (ryanthemanuel) [reply packages/driver/src/cypress/proxy-logging.ts]: Fixed here: [`70b07df` (#21373)](https://github.com/cypress-io/cypress/pull/21373/commits/70b07dfe4acb71861a97c01660c3017e82aae5b1)

**Review by mschile — APPROVED** [review]

File: packages/driver/src/cypress/proxy-logging.ts [hunk 3]
```diff
@@ -343,6 +337,22 @@ export default class ProxyLogging {
...
 
+  private updateRequestWithResponse (responseReceived: BrowserResponseReceived): void {
+    const proxyRequest = _.find(this.proxyRequests, ({ preRequest }) => preRequest.requestId === responseReceived.requestId)
+
+    if (!proxyRequest) {
+      return debug('unmatched responseReceived event %o', responseReceived)
+    }
+
+    if (proxyRequest.xhr && proxyRequest.xhr.xhr.readyState !== XMLHttpRequest.DONE) {
+      proxyRequest.xhr.xhr.addEventListener('load', () => {
```
Comment (flotwig) [inline packages/driver/src/cypress/proxy-logging.ts:348]: :+1:

**Review by flotwig — APPROVED** [review]

**Pull request merged**